### PR TITLE
[Feature] Implemented Sign Up Form Validator

### PR DESCRIPTION
**Pull request by StanleyOned**
[Feature] Implemented Sign Up Form Validator.
- Email text validator
- Password validation with different types.

### CODE REVIEW

File: DevKit/DevKit/Classes/Authentication/FormValidator.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import Foundation
+
+/// FormValidator used to validate sign up forms.
+public class FormValidator {
+    
+    /// Validates the password with the given validation types and minimum Characters.
+    /// - Parameters:
+    ///   - password: The email given.
+    ///   - passwordType: The type of password.
+    public static func isValid(_ password: String, validators: [ValidationType], minCharacters: Int) -> Bool {
```
Comment (hkellaway) [inline DevKit/DevKit/Classes/Authentication/FormValidator.swift]: @jgsamudio as mentioned to @StanleyOned, i tend to prefer instance functions to static ones - as static encourages the client to enact tight coupling / adds a barrier to creating a mock

**Comment by hkellaway** [comment]
tests are vital to open source projects @StanleyOned - and in this case ought to be fairly straightforward / help you identify edge cases / describe precisely for clients the breadth of what FormValidator does

**Comment by hkellaway** [comment]
keeping the README and CHANGELOG up-to-date are also vital for open source projects @StanleyOned - pls make sure to understand with @jgsamudio whether that is something you are expected to update per Pull Request or if the updates are done when a new release is being prepared

File: DevKit/DevKit/Classes/Authentication/FormValidator.swift (hunk 1 above)
Comment (jgsamudio) [reply DevKit/DevKit/Classes/Authentication/FormValidator.swift]: @hkellaway Using static closely resembles something that can be found in an extension function because the class does not need to store any data. However an instance will be good for mocking while testing.

**Comment by ThibaultKlein** [comment]
Have you looked into [GenericValidator](https://github.com/prolificinteractive/GenericValidator)? Not trying to self-promote this library but I feel we solved this issue already and it might be more beneficial to consolidate our effort on something that already exists if possible.

Can DevKit have a dependency on GenericValidator and in DevKit you make available pre-defined validators like an email/password one for example?

**Comment by StanleyOned** [comment]
> tests are vital to open source projects @StanleyOned - and in this case ought to be fairly straightforward / help you identify edge cases / describe precisely for clients the breadth of what FormValidator does

Added some Unit Test 👍 @hkellaway

File: DevKit/DevKit/Classes/Authentication/ValidationType.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            case .uppercasedLetters:
+                regexCode += ValidationType.uppercasedLetters.rawValue
+            case .numbers:
+                regexCode += ValidationType.numbers.rawValue
+            case .specialCharacters:
+                regexCode += ValidationType.specialCharacters.rawValue
+            }
+        }
+        regexCode += ".{\(minCharacters),}$"
+        return regexCode
```
[the remark this answers is not in the document]
Comment (StanleyOned) [reply DevKit/DevKit/Classes/Authentication/ValidationType.swift]: Yea you can, their also an Or regex code but we haven't used it yet for this scenario. I'll actually look into it to see if I can refactor some of the code.

File: DevKit/DevKit/Classes/Validation/ValidationType.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+//
+
+/// Enum to determine the password type.
+///
+/// **Subspec: Validation/ValidationType**
+///
+/// - lowercasedLetters
+/// - uppercasedLetters
+/// - specialCharacters
+/// - numbers
```
Comment (jgsamudio) [inline DevKit/DevKit/Classes/Validation/ValidationType.swift]: Add min characters
